service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java

```diff
@@ -1604,7 +1612,7 @@ private JobProperties ProcessJobResponse(HttpResponse response) throws IotHubExc
 
     private HttpRequest CreateRequest(URL url, HttpMethod method, byte[] payload, String sasToken) throws IOException
     {
-        HttpRequest request = new HttpRequest(url, method, payload, proxy);
+        HttpRequest request = new HttpRequest(url, method, payload, this.options != null && this.options.getProxyOptions() != null ? this.options.getProxyOptions().getProxy() : null);
```

Comment:
there is a whole lot happening here, can we move this to the line above:
this.options != null && this.options.getProxyOptions() != null ? this.options.getProxyOptions().getProxy() : null

Reply:
The way I see, registry manager options should never be null, but have some default initial value (proxy = null)

Reply:
Yeah, good call. This is pretty busy